src/module/migration/runner/base.ts

```diff
@@ -79,9 +78,11 @@ export class MigrationRunnerBase {
         for await (const migration of migrations) {
             await migration.updateItem?.(current);
             // Handle embedded spells
-            if (current.type === "consumable" && current.system.spell.data) {
-                this.preMigrateEmbeddedSource(current.system.spell.data);
-                await migration.updateItem?.(current.system.spell.data);
+            if (current.type === "consumable") {
+                await this.preMigrateEmbeddedSpell(current);
```

Comment:
This runs once for every migrations which could be a lot depending on the current world system schema version